### DISCUSSION TITLE
Variant context

### DISCUSSION THREAD

**Discussion by leroyvn** [mitsuba-renderer/mitsuba3#699] [Feature requests]
Dear Mitsuba maintainers, I sometimes find myself in a situation where a temporary variant override could be useful. A context manager would be great in such cases:
```python
mi.set_variant("scalar_rgb")

with mi.variant_context("llvm_rgb"):
    # Do some preprocessing faster with the LLVM backend
    ...

mi.render(my_scene)
```
Would you consider such a feature?

**Comment by njroussel** [comment]
I really like this and genuinely thought we already had this (I was surprised when my script failed). I'd welcome a PR for it!

**Reply by leroyvn** [reply]
I opened #700.
